backend/substrapp/ledger/api.py

```diff
@@ -168,7 +168,19 @@ def call_ledger(channel_name, call_type, fcn, *args, **kwargs):
     ts = time.time()
     error = None
     try:
-        return _call_ledger(channel_name, call_type, fcn, *args, **kwargs)
+        response = _call_ledger(channel_name, call_type, fcn, *args, **kwargs)
+
+        if isinstance(response, dict) and 'bookmark' in response:
+            results = response['results']  # first results
+            while response['results'] and len(response['bookmark']) > 0:
```

Comment:
```suggestion
            while len(response['bookmark']) > 0:
```
should be suffisent, no?

Reply:
Nope, `response['results']` should not be empty as, if the query returns, at least, a bookmark, it will always return a bookmark (end bookmark), even if `response['results']` is empty.

Reply:
that make sens, thank's